### DISCUSSION TITLE
"Fatal" warning during MOOSE update

### DISCUSSION THREAD

**Discussion by EdSheltonUKAEA** [idaholab/moose#26465] [Q&A Modules: General] [answered]
I'm trying to do a Conda/Mamba and MOOSE update and got the following when attempting to do the git rebase:

> (moose) eshelton@K0214:~$ cd ~/projects/moose
> (moose) eshelton@K0214:~/projects/moose$ git fetch origin
> remote: Enumerating objects: 19833, done.
> remote: Counting objects: 100% (12117/12117), done.
> remote: Compressing objects: 100% (2079/2079), done.
> remote: Total 19833 (delta 10437), reused 11338 (delta 9968), pack-reused 7716
> Receiving objects: 100% (19833/19833), 17.08 MiB | 8.36 MiB/s, done.
> Resolving deltas: 100% (14937/14937), completed with 1892 local objects.
> From https://github.com/idaholab/moose
>    c9f69be921..c5bcc5d78c  master                   -> origin/master
>  * [new branch]            2023-11-08-release       -> origin/2023-11-08-release
>  * [new branch]            assembly-refactors-25666 -> origin/assembly-refactors-25666
>    e8c7f090d9..c835ce5787  devel                    -> origin/devel
>    9f83051c5f..3833a5c799  next                     -> origin/next
>    c9f69be921..c5bcc5d78c  scheduled                -> origin/scheduled
>  * [new branch]            thm-joss-paper-review    -> origin/thm-joss-paper-review
>  * [new tag]               2023-11-08               -> 2023-11-08
> (moose) eshelton@K0214:~/projects/moose$ git rebase origin/master
> First, rewinding head to replay your work on top of it...
> error: Updating the following directories would lose untracked files in them:
>         modules/heat_conduction/src
> 
> Aborting
> fatal: Could not detach HEAD
> (moose) eshelton@K0214:~/projects/moose$

The "Aborting" and "fatal" messages don't look like good news!
I don't think I've ever had a problem like this before running the git fetch and rebase commands.
Has the MOOSE update procedure changed, or is there a problem here.
Is it safe for me to proceed with the clobberall and make in my project directory, or is this likely to fail if the git update didn't work?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Did you put work in the heat conduction module?

That module was renamed to heat transfer. The rebase should have worked but maybe it got confused because of the sym links we now have in the former heat conduction module.

How many commits are we talking about here? Could we consider doing this manually?

Is this branch public?

Guillaume

**Reply by EdSheltonUKAEA** [reply]
Hello Guillaume - and Happy New Year!

It's possible some custom files have been added, but I don't think so.  I'm pretty sure any new kernels, etc. have been added in my application rather than the overarching MOOSE folder.

I don't have a module called heat transfer, so I assume haven't done an update since before this change was implemented.
I've got a more recent installation on a different computer, so I'll take a look at that to so if that has the heat transfer module instead.

I'm not sure what the question about commits means?  In the block of text it says "Enumerating obects: 19833", etc.

The branch isn't public, or at least shouldn't be.

Ed

**Reply by EdSheltonUKAEA** [reply]
On the other PC with the more recent MOOSE installation, the git commands work OK.  This installation has both folders:
~projects\moose\modules\heat_conduction
~projects\moose\modules\heat_transfer

Then in my application folder (~/projects/eshelton/red-kite) when running make clobberall or make -j8, I am prompted with:

> /projects/moose//modules/modules.mk:80: The heat conduction module was renamed to the heat transfer module. Please update your Makefile and replace HEAT_CONDUCTION with HEAT_TRANSFER

So I've done this by adding this line to Makefile.

> HEAT_TRANSFER		    := yes

Now "make clobberall" seems to run OK.
But "make -j8" throws up this error a couple of times before stopping:

> Compiling C++ (in opt mode) /home/timodel01/projects/moose//framework/contrib/exodiff/exo_block.C...
> /home/timodel01/projects/moose//framework/contrib/hit/parse.cc: In member function 'hit::Field::Kind hit::Field::kind()':
> /home/timodel01/projects/moose//framework/contrib/hit/parse.cc:687:34: error: 'class wasp::HITNodeView' has no member named 'token_type'; did you mean 'set_type'?
>   687 |     auto token_type = value_node.token_type();
>       |                                  ^~~~~~~~~~
>       |                                  set_type
> Compiling C++ (in opt mode) /home/timodel01/projects/moose//framework/contrib/exodiff/SL_tokenize.C...
> Compiling C++ (in opt mode) /home/timodel01/projects/moose//framework/contrib/exodiff/FileInfo.C...
> Compiling C++ (in opt mode) /home/timodel01/projects/moose//framework/contrib/exodiff/add_to_log.C...
> Compiling C++ (in opt mode) /home/timodel01/projects/moose//framework/contrib/exodiff/util.C...
> Compiling C++ (in opt mode) /home/timodel01/projects/moose//framework/contrib/exodiff/stringx.C...
> Compiling C++ (in opt mode) /home/timodel01/projects/moose//framework/contrib/exodiff/check.C...
> Compiling C++ (in opt mode) /home/timodel01/projects/moose//framework/contrib/exodiff/map.C...
> make: *** [/home/timodel01/projects/moose//framework/build.mk:135: /home/timodel01/projects/moose//framework/contrib/hit/parse.x86_64-conda-linux-gnu.opt.lo] Error 1
> make: *** Waiting for unfinished jobs....
> /home/timodel01/projects/moose//framework/contrib/hit/parse.cc: In member function 'hit::Field::Kind hit::Field::kind()':
> /home/timodel01/projects/moose//framework/contrib/hit/parse.cc:687:34: error: 'class wasp::HITNodeView' has no member named 'token_type'; did you mean 'set_type'?
>   687 |     auto token_type = value_node.token_type();
>       |                                  ^~~~~~~~~~
>       |                                  set_type
> make: *** [/home/timodel01/projects/moose//framework/moose.mk:138: /home/timodel01/projects/moose//framework/contrib/hit/hit.so] Error 1

It looks like this could be something different from the heat_transfer module, maybe?

**Reply by GiudGiud** [reply]
Did you make cleanall in between both builds? Having objects from the two modules won't help

I was asking about the number of commits that you added on top of your previous moose branch. If not too many, it would not be difficult to move them manually

So are you all set using a new version of moose without rebasing?

**Reply by GiudGiud** [reply]
ok the errors are unrelated to heat transfer / heat conduction actually

```
/home/timodel01/projects/moose//framework/contrib/hit/parse.cc:687:34: error: 'class wasp::HITNodeView' has no member named 'token_type'; did you mean 'set_type'?
687 | auto token_type = value_node.token_type();
```
this is from WASP. You ll want to update your conda packages if you are using that.
If not then you ll need to run update_and_rebuild_wasp

**Reply by EdSheltonUKAEA** [reply]
I've done both "conda update --all" and "mamba update --all".  It still results in the same error when trying to run make in my applications folder.  
How and in what folder should I run update_and_rebuild_wasp?

**Reply by GiudGiud** [reply]
I d delete mamba and re-install with conda. 

We actually made the switch from mambaforge to miniforge so it would not be surprising to have difficulties download new packages

**Reply by EdSheltonUKAEA** [reply]
OK.  Thanks.  I'll work with colleagues who know more about the software operating systems behind MOOSE to try to resolve these problems.

**Reply by GiudGiud** [reply]
> How and in what folder should I run update_and_rebuild_wasp?

This should be done in the moose/scripts folder. But not if you are using conda. If you are using conda, the conda-downloaded packages will include moose-wasp.

If they dont, the conda update is failing. And the easiest way to fix it are either a new conda environment or simply wipe conda and install it again. Since we also transitioned from mambaforge to miniforge, I would recommend the latter

**Reply by milljm** [reply]
If you choose to reinstall Conda, please be sure to re-follow our instructions: https://mooseframework.inl.gov/getting_started/installation/conda.html

There is a one-liner folks seems to forget: `conda update --all` immediately after installing Conda. This is necessary to get the very latest actual version of `conda` which has the correct package solver in it.

**Reply by EdSheltonUKAEA** [reply]
Hi - I was able to get a colleague to help me do a complete removal and reinstall of MOOSE and the supporting software package management tools on both my laptop and shared desktop, so this is all sorted now.  Thanks again for the advice and help.
Ed
